File: spring-backend/src/main/java/com/vizagsteel/inventory/service/SupplierService.java
package com.vizagsteel.inventory.service;

import com.vizagsteel.inventory.entity.Supplier;
import com.vizagsteel.inventory.repository.SupplierRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class SupplierService {

    private final SupplierRepository supplierRepository;

    public SupplierService(SupplierRepository supplierRepository) {
        this.supplierRepository = supplierRepository;
    }

    public List<Supplier> getAllSuppliers() {
        return supplierRepository.findAll();
    }

    public Supplier saveSupplier(Supplier supplier) {
        return supplierRepository.save(supplier);
    }

    public Supplier getSupplierById(Long id) {
        return supplierRepository.findById(id).orElseThrow(() -> new IllegalArgumentException("Supplier not found"));
    }

    public void deleteSupplier(Long id) {
        supplierRepository.deleteById(id);
    }
}

File: backend/src/services/supplierService.js
const supplierModel = require('../models/supplierModel');

exports.getAllSuppliers = async () => supplierModel.findAll();
exports.getSupplierById = async (id) => supplierModel.findById(id);
exports.createSupplier = async (supplier) => supplierModel.create(supplier);
exports.updateSupplier = async (id, supplier) => supplierModel.update(id, supplier);
exports.deleteSupplier = async (id) => supplierModel.deleteById(id);
